manager/apis/app/v1alpha1/conditions_types.go

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+// ConditionType represents a condition type of m4dapplication and blueprint resources
+type ConditionType string
+
+const (
+	// ReadyCondition means that data endpoints are available for usage
+	ReadyCondition ConditionType = "Ready"
+
+	// ErrorCondition means that an error was encountered, e.g. failure to establish communication with an external service
+	// TODO: add a number of retrials, not activate failure immediately
+	ErrorCondition ConditionType = "Error"
+
+	// FailureCondition represents a critical failure, e.g. denial of data access or failure of an allocated resource
+	FailureCondition ConditionType = "Failure"
+
+	// TerminatingCondition means that deletion is in progress
+	TerminatingCondition ConditionType = "Terminating"
+)
```

Comment:
having both Error and Failure is misleading you should merge them. The can be used to provide info about the exact cause

Reply:
> 
> 
> having both Error and Failure is misleading you should merge them. The can be used to provide info about the exact cause

I prefer not to do this change in this PR since I'm not introducing a new type. Condition structure was present in m4dapplication_types, in this PR it is reused for blueprint_types as well. The existing logic handles both condition types. This was done to distinguish between a failure and an error that could disappear on further retrials. I will open another issue for error management that will attempt several retrials and raise Failure if all failed.

Reply:
ok, please provide a link to the new issue and we cab resolve this thread